integrations/stripe/src/misc/custom-schemas.ts

```diff
@@ -0,0 +1,241 @@
+import z from 'zod'
+import {
+  createPaymentLinkUi,
+  createSubsLinkUi,
+  createCustomerUi,
+  deactivatePaymentLinkUi,
+  findPaymentLinkUi,
+  listCustomersUi,
+  retrieveCustomerByIdUi,
+  searchCustomersUi,
+} from './custom-uis'
+import { partialCustomer } from './sub-schemas'
+
+export const createPaymentLinkInputSchema = z.object({
+  productName: z.string().describe(createPaymentLinkUi.productName.title),
+  unit_amount: z
+    .number()
+    .optional()
+    .default(0)
+    .describe(createPaymentLinkUi.unit_amount.title),
+  currency: z
+    .string()
+    .optional()
+    .default('usd')
+    .describe(createPaymentLinkUi.currency.title),
+  quantity: z
+    .number()
+    .optional()
+    .default(1)
+    .describe(createPaymentLinkUi.quantity.title),
+  adjustableQuantity: z
+    .boolean()
+    .optional()
+    .default(false)
+    .describe(createPaymentLinkUi.adjustableQuantity.title),
+  adjustableQuantityMaximum: z
+    .number()
+    .min(2)
+    .max(999)
+    .optional()
+    .default(99)
+    .describe(createPaymentLinkUi.adjustableQuantityMaximum.title),
+  adjustableQuantityMinimum: z
+    .number()
+    .min(1)
+    .max(998)
+    .optional()
+    .default(1)
+    .describe(createPaymentLinkUi.adjustableQuantityMinimum.title),
+})
+
+export const createPaymentLinkOutputSchema = z
+  .object({
+    id: z.string(),
+    url: z.string(),
+  })
+  .partial()
+
+export const listProductPricesInputSchema = z.object({})
+
+export const listProductPricesOutputSchema = z
+  .object({
+    products: z.record(
+      z.object({
+        name: z.string(),
+        prices: z.array(
+          z.object({
+            unit_amount: z.number().nullable(),
+            currency: z.string(),
+            recurring: z.object({}).passthrough().nullable().optional(),
+          })
+        ),
+      })
+    ),
+  })
+  .partial()
+
+export const createSubsLinkInputSchema = z.object({
+  productName: z.string().describe(createSubsLinkUi.productName.title),
+  unit_amount: z
+    .number()
+    .optional()
+    .default(0)
+    .describe(createSubsLinkUi.unit_amount.title),
+  currency: z
+    .string()
+    .optional()
+    .default('usd')
+    .describe(createSubsLinkUi.currency.title),
+  quantity: z
+    .number()
+    .optional()
+    .default(1)
+    .describe(createSubsLinkUi.quantity.title),
+  adjustableQuantity: z
+    .boolean()
+    .optional()
+    .default(false)
+    .describe(createSubsLinkUi.adjustableQuantity.title),
+  adjustableQuantityMaximum: z
+    .number()
+    .min(2)
```

Comment:
**question:** where do all these constants come from ?

Reply:
Stripe documentation in its typings specifies it: The maximum quantity the customer can purchase. By default this value is 99. You can specify a value up to 999.